packages/webdriverio/src/utils.ts

```diff
@@ -184,3 +185,15 @@ export const axeFinishRun = ({
     `)
   );
 };
+
+export async function axeShadowSelect(
+  client: BrowserObject,
+  selector: CrossTreeSelector
+): Promise<Element | undefined> {
+  const selectors: string[] = Array.isArray(selector) ? selector : [selector];
+  let elmRef: Element | undefined = await client.$(selectors[0]);
+  for (let i = 1; i < selectors.length && elmRef; i++) {
+    elmRef = await client.$(selectors[i]);
```

Comment:
This does not enter shadow dom, it only looks at light dom. Moreover, it grabs the selector from the top-level `client` and does not use the last `elmRef` as a starting point. So this selector`['#shadow-root', 'iframe']` will find the first `iframe` element on the page, not the iframe that is a child of `#shadow-root`.